db/data/20240522123948_update_modern_language_child_subjects.rb

```diff
@@ -0,0 +1,55 @@
+class UpdateModernLanguageChildSubjects < ActiveRecord::Migration[7.1]
+  def up
+    api_response = PublishTeacherTraining::Subject::Api.call
+    child_subject_ids = modern_language_child_subject_ids(api_response.fetch("data"))
+
+    subjects_data = api_response.fetch("included")
+
+    assign_parent_subject(parent_subject: modern_languages, child_subject_ids:, subjects_data:)
+  end
+
+  def down
+    return if modern_languages.blank?
+
+    Subject.where(parent_subject: modern_languages).update_all(parent_subject_id: nil)
```

Comment:
Strong migrations will prevent us from using our app models in migrations, as when the code is eager-loaded, previous definitions will break.

A recommendation is to create a migration-scoped AR class.

```ruby
class SomeMigration
  class SubjectForMigration
    self.table_name = "subjects"
  end

  def up
    SubjectForMigration.where(...).update_all(...)
  end
end
```

This way, app changes don't break migrations, they're self-contained.

Reply:
This is not a migration. This is a [data-migration](https://github.com/ilyakatz/data-migrate) and strong migration does not apply.
This is also a point in time migration of data, and will likely be removed after a certain amount of time has past.